execution_engine/src/shared/host_function_costs.rs

```diff
@@ -0,0 +1,493 @@
+        Self {
+            cost,
+            arguments: Vec::new(),
+        }
+    }
+}
+
+impl Distribution<HostFunction> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> HostFunction {
+        HostFunction {
+            cost: rng.gen(),
+            arguments: Vec::new(),
+        }
+    }
+}
+
+impl ToBytes for HostFunction {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut ret = bytesrepr::unchecked_allocate_buffer(self);
+        ret.append(&mut self.cost.to_bytes()?);
+        ret.append(&mut self.arguments.to_bytes()?);
+        Ok(ret)
+    }
+
+    fn serialized_length(&self) -> usize {
+        self.cost.serialized_length() + self.arguments.serialized_length()
+    }
+}
+
+impl FromBytes for HostFunction {
+    fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
+        let (cost, bytes) = FromBytes::from_bytes(bytes)?;
+        let (arguments, bytes) = FromBytes::from_bytes(bytes)?;
+        Ok((Self { cost, arguments }, bytes))
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug, DataSize)]
+pub struct HostFunctionCosts {
+    pub read_value: HostFunction,
+    pub read_value_local: HostFunction,
+    pub write: HostFunction,
+    pub write_local: HostFunction,
+    pub add: HostFunction,
+    pub add_local: HostFunction,
+    pub new_uref: HostFunction,
+    pub load_named_keys: HostFunction,
+    pub ret: HostFunction,
+    pub get_key: HostFunction,
+    pub has_key: HostFunction,
+    pub put_key: HostFunction,
+    pub remove_key: HostFunction,
+    pub revert: HostFunction,
+    pub is_valid_uref: HostFunction,
+    pub add_associated_key: HostFunction,
+    pub remove_associated_key: HostFunction,
+    pub update_associated_key: HostFunction,
+    pub set_action_threshold: HostFunction,
+    pub get_caller: HostFunction,
+    pub get_blocktime: HostFunction,
+    pub create_purse: HostFunction,
+    pub transfer_to_account: HostFunction,
+    pub transfer_from_purse_to_account: HostFunction,
+    pub transfer_from_purse_to_purse: HostFunction,
+    pub get_balance: HostFunction,
+    pub get_phase: HostFunction,
+    pub get_system_contract: HostFunction,
+    pub get_main_purse: HostFunction,
+    pub read_host_buffer: HostFunction,
+    pub create_contract_package_at_hash: HostFunction,
+    pub create_contract_user_group: HostFunction,
+    pub add_contract_version: HostFunction,
+    pub disable_contract_version: HostFunction,
+    pub call_contract: HostFunction,
+    pub call_versioned_contract: HostFunction,
+    pub get_named_arg_size: HostFunction,
+    pub get_named_arg: HostFunction,
+    pub remove_contract_user_group: HostFunction,
+    pub provision_contract_user_group_uref: HostFunction,
+    pub remove_contract_user_group_urefs: HostFunction,
+    pub print: HostFunction,
+}
+
+impl Default for HostFunctionCosts {
+    fn default() -> Self {
+        Self {
+            read_value: HostFunction::fixed(0),
```

Comment:
Could we not just add a line:
```rust
let fixed = HostFunction::fixed(DEFAULT_FIXED_COST)
```
and then use that in place of all of these explicit constructions (and use constant instead of magic numbers).

Reply:
Done - I went with `impl Default for HostFunction` that uses `DEFAULT_FIXED_COST` and then since legitimate HostFunctionCosts defaults aren't wired yet I just relied on derived `Default` trait in `HostFunctionCosts`

Reply:
:+1: